### ISSUE TITLE
Singleton scope not honoured when obtaining view controller from storyboard

### ISSUE THREAD

**Issue by rhoberman** [appsquickly/Typhoon#374]
Following the docs here (https://github.com/appsquickly/Typhoon/wiki/Storyboards) I am able to inject objects into view controllers obtained via the Storyboard (nice!).

Although I scoped my "Singleton" class appropriately each view controller receives a separate instance.

I've created a project demonstrating the problem here: https://github.com/rhoberman/TyphoonSingletonStoryboard

The singleton scope seems to be having an effect: if I inject the singleton twice into the same controller I get the same instance.

Is this a bug or am I missing something?

**Comment by alexgarbarev** [comment]
@rhoberman Thanks for feedback.

Yes, there is bug, but in another place.

Your mistake here:
```swift
let mainStoryboard: UIStoryboard = UIStoryboard(name: "Main", bundle: nil)
var nextViewController = mainStoryboard.instantiateViewControllerWithIdentifier("tabBarController") as! UIViewController
```
The problem that you create UIStoryboard instance again. Why you re-create UIStoryboard again, instead of ```self.storyboard```?

My fault is that you've got ``TyphoonStoryboard`` instance instead of ``UIStoryboard``, so after I'll fix that bug, your code will not inject dependencies into ```FirstTabViewController```, because it's created by ``UIStoryboard``

Is everything clear now?

**Comment by alexgarbarev** [comment]
Bug was fixed in ``3.2.1``.

---

Some additional documentation for you:

1) To inject dependencies into ``UIViewControllers`` in Storyboard automatically, you should use ``TyphoonStoryboard`` instead of ``UIStoryboard``. ``TyphoonStoryboard`` is ``UIStoryboard`` subclass which injects dependencies in each instantiated viewController. ``TyphoonStoryboard`` resolves dependecies using ``TyphoonComponentFactory`` object which comes within initializer. Without that object, ``TyphoonStoryboard`` is useless, because haven't source to resolve dependencies.

2)  In Typhoon terms, there is `initial factory` which is created from your plist automatically. If factory from plist created successfully, then instead of initial ``UIStoryboard`` which name specified as ``UIMainStoryboardFile``, ``TyphoonStoryboard`` is created with that factory. This ``initial factory`` creates only once, during application startup. If you want to create another storyboard and keep singletons from `initial factory`, you should create another ``TyphoonStoryboard`` instance and passthrough ``initial factory`` into it

3) All singletons lives in factory that created them. If you'll create another instance of ``TyphoonComponentFactory`` it will have own singletons pool. Keep one ``TyphoonComponentFactory`` instance and use across the app

---

Now, what to do with your app, to fix the problem:
1) ``pod update``
2) Confirm it's crashes now, with `nil` dependencies (so, my bugfix working)
3) Replace 
```swift
let mainStoryboard: UIStoryboard = UIStoryboard(name: "Main", bundle: nil)
var nextViewController = mainStoryboard.instantiateViewControllerWithIdentifier("tabBarController") as! UIViewController 
```
with
```swift
var nextViewController = self.storyboard!.instantiateViewControllerWithIdentifier("tabBarController") as! UIViewController
```
4) Confirm, it's working now

**Comment by rhoberman** [comment]
@alexgarbarev Thank you so much for your quick and detailed response.

I can confirm that after upgrading to 3.2.1 it crashes with nil dependencies and that my demo code works when I use self.storyboard.
